addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,193 @@
+        return Constant.messages.getString("ascanalpha.forbiddenBypass.name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("ascanalpha.forbiddenBypass.desc");
+    }
+
+    @Override
+    public void scan() {
+        HttpMessage message = getNewMsg();
+        try {
+            sendAndReceive(message);
+            if (message.getResponseHeader().getStatusCode() == HttpStatusCode.FORBIDDEN) {
+                String FULL_URL = message.getRequestHeader().getURI().toString();
+                URI uri = new URI(FULL_URL);
+                String HOST = uri.getAuthority();
+                String PATH = uri.getPath();
+                String SCHEMA = uri.getScheme();
+                List<String> upayloads =
+                        new LinkedList<String>(
+                                Arrays.asList(
+                                        new String[] {
+                                            "/%2e" + PATH,
+                                            PATH + "/.",
+                                            "/." + PATH + "/./",
+                                            PATH + "%20/",
+                                            "/%20" + PATH + "%20/",
+                                            PATH + "..;",
+                                            PATH + "?",
+                                            PATH + "??",
+                                            "/" + PATH + "//",
+                                            PATH + "/",
+                                            PATH + "/.testus",
+                                            PATH + "../app.py"
+                                        }));
+                List<String> hpayloads =
+                        new LinkedList<String>(
+                                Arrays.asList(
+                                        new String[] {
+                                            "X-Rewrite-URL: " + PATH,
+                                            "X-Original-URL: " + PATH,
+                                            "Referer: " + PATH,
+                                            "X-Custom-IP-Authorization: 127.0.0.1",
+                                            "X-Originating-IP: 127.0.0.1",
+                                            "X-Forwarded-For: 127.0.0.1",
+                                            "X-Remote-IP: 127.0.0.1",
+                                            "X-Client-IP: 127.0.0.1",
+                                            "X-Host: 127.0.0.1",
+                                            "X-Forwared-Host: 127.0.0.1"
+                                        }));
+
+                // sending payloads in URL.
+                for (String i : upayloads) {
+                    String TMP_URI = SCHEMA + "://" + HOST + "" + i;
+                    HttpMessage req_with_payload =
+                            new HttpMessage(new org.apache.commons.httpclient.URI(TMP_URI, true));
```

Comment:
Should just use this URI form, and drop the use of the Java one.

Reply:
Hi @kingthorin, Which URI form you are mentioning here? tmpUri?

Reply:
I'd suggest sticking with the Apache one, and not using the Java one.

The reason you had to fully quality the declaration here is that you've also imported the Java URI class, if you're only using one then there's fewer issues.